src/auth/AuthManager.ts

```diff
@@ -167,11 +197,34 @@ export class AuthManager {
 
     return Promise.resolve({
       isLoggedIn,
-      userName: userName?.toLowerCase(),
+      userName: userName.toLowerCase(),
       loginForm
     })
   }
 
+  private async fetchUserName(): Promise<{
+    isLoggedIn: boolean
+    userName: string
+  }> {
+    //For VIYA we will send request on API endpoint. Which is faster then pinging SASJobExecution.
+    //For SAS9 we will send request on SASStoredProcess
+    const url =
+      this.serverType === 'SASVIYA'
```

Comment:
We should check against the `ServerType` enum from `@sasjs/utils/types`.

Reply:
fixed.